src/platform_impl/linux/wayland/window/shim.rs

```diff
@@ -188,6 +214,52 @@ impl WindowHandle {
         }
     }
 
+    pub fn set_user_attention(&self, request_type: Option<UserAttentionType>) {
+        let xdg_activation = match self.xdg_activation.as_ref() {
+            None => return,
+            Some(xdg_activation) => xdg_activation,
+        };
+
+        // Unsetting is done by compositor.
+        if request_type.is_none() {
+            return;
+        }
```

Comment:
Could in theory combine this with the other match statement, that way only a single `return` would be necessary which might make it easier to read for some. But I personally also like just keeping the conditional logic trivial and separate so this should be fine.